addons/website_mail/controllers/main.py

```diff
@@ -1,12 +1,18 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
-from hashlib import sha1
-from time import time
+
 from werkzeug.exceptions import NotFound
 
 from odoo import http
 from odoo.http import request
 
+
+def _special_access_object(res_model, res_id, token='', token_field=''):
+    record = request.env[res_model].browse(res_id)
+    if token and record and token == getattr(record.sudo(), token_field, None):
```

Comment:
Ok, but it have to check than the token_field is filled to avoid given None as argument to consteq. So the condition looks like : 
`if token and record and getattr(record.sudo(), token_field, None) and consteq(getattr(record.sudo(), token_field, None), str(token)):`